sdk/trace/trace_test.go

```diff
@@ -1164,6 +1164,59 @@ func TestRecordErrorNil(t *testing.T) {
 	}
 }
 
+func TestRecordErrorWithStatus(t *testing.T) {
+	scenarios := []struct {
+		err error
+		typ string
+		msg string
+	}{
+		{
+			err: ottest.NewTestError("test error"),
+			typ: "go.opentelemetry.io/otel/internal/internaltest.TestError",
+			msg: "test error",
```

Comment:
Should we add a test case that the error is nil?

Reply:
Done, thank you very much.